docs/api/webxrHardwareExtensionAPIs.md

```diff
@@ -1,16 +1,42 @@
----
+.---
 title: WebXR Hardware Extension APIs
 type: api
 layout: docs
 order: 2
 parent_section: api
 ---
 
-This currently consists of seperate device-specific APIs, but will be folded into a more generic API, properly coded in Exokit generically on the `session` objects rather than under a `magicleap` namespace.
+The WebXR Hardware Extension APIs is a generic API that is not tied to any one hardware platform and provides access to: meshing, planes detection, hand tracking, eye tracking. This is inspired by the [WebExtension API style](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API). Although it is not a web standard yet, these are finding their way into the WebXR spec and Exokit wants to be standards-compliant.
+
+No hardware is needed to test these APIs as Exokit Studio provides emulation and emulated data.
+
+# Basic usage
+
+Request a `session` instance with the `display.requestSession()` method in order to interact with XR device's presentation or tracking capabilities. Specify under `extensions` which APIs you want to access:
+
+```js
+const session = await display.requestSession({
+  exclusive: true,
+  extensions: {
+    meshing: true,
+    planesTracking: true,
+    handTracking: true,
+    eyeTracking: true,
+  },
+});
+  session.addEventListener('meshadd', _meshadd);
+  session.addEventListener('meshupdate', _meshupdate);
+  session.addEventListener('meshremove', _meshremove);
+```
+
+When the API mechanism for an `InputSource` is invoked, events are fired on `session`.
```

Comment:
Each one of the extensions behaves differently.

Meshing for example emits update events on the `XRSession`. The shape of these needs to be documented.

Planes works similarly.

Hands and eye tracking however exposes the hands and eye (single eye with axes for blink) as additional `InputSource`s. This is intended to be as compliant as possible with the WebXR spec.

Reply:
I'll update instructions per `InputSource` to cover the different cases (planes, meshing, hands, eyes).

Reply:
Added basic documentation per `InputSources` in d586984 , not sure if they should be further documented with example code in this doc.